library/core/lifecycle-events/src/main/java/org/quiltmc/qsl/lifecycle/api/client/event/ClientTickEvents.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.lifecycle.api.client.event;
+
+import org.quiltmc.qsl.base.api.event.ArrayEvent;
+
+import net.minecraft.client.MinecraftClient;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+
+/**
+ * Events indicating progress through the tick loop of a Minecraft client.
+ *
+ * <h2>A note of warning</h2>
+ *
+ * Callbacks registered to any of these events should ensure as little time as possible is spent executing, since the tick
+ * loop is a very hot code path.
+ */
+@Environment(EnvType.CLIENT)
+public final class ClientTickEvents {
+	/**
+	 * An event indicating an iteration of the client's tick loop will start.
+	 */
+	public static final ArrayEvent<Start> START = ArrayEvent.create(Start.class, callbacks -> client -> {
```

Comment:
Would `BEGIN` be a better name as its the beginning of the tick? also prefixing with `_TICK` might help differentiate between this even and the world start event

Reply:
I mean, we also can say that about `START` as it's the start of the tick ;)
Not sure that there's really a need to change those.

And I don't think prefixing here is useful since the class name already does the job.

Reply:
I deemed the prefix/suffix to be redundant considering the name of the class.